test/integ/pandas/test_arrow_pandas.py

```diff
@@ -984,3 +1053,40 @@ def test_resultbatches_pandas_functionality(conn_cnx):
     )
     final_df = pyarrow.concat_tables(tables).to_pandas()
     assert numpy.array_equal(expected_df, final_df)
+
+
+@pytest.mark.skipif(
+    not installed_pandas or no_arrow_iterator_ext,
+    reason="arrow_iterator extension is not built, or pandas is missing.",
+)
+def test_pandas_telemetry(conn_cnx, capture_sf_telemetry):
+    cases = ["NULL", 0.11, -0.11, "NULL", 1.27, -1.28, "NULL"]
+    table = "test_arrow_tiny_int"
+    column = "(a number(5,2))"
+    values = (
+        "(" + "),(".join(["{}, {}".format(i, c) for i, c in enumerate(cases)]) + ")"
+    )
+    init(conn_cnx, table, column, values)
+    sql_text = "select a from {} order by s".format(table)
+
+    validate_pandas_with_telemetry(
+        conn_cnx,
+        capture_sf_telemetry,
+        sql_text,
+        cases,
+        1,
+        "one",
+        telemetry_to_check=TelemetryField.PANDAS_FETCH_ALL,
+    )
+
+    validate_pandas_with_telemetry(
+        conn_cnx,
+        capture_sf_telemetry,
+        sql_text,
+        cases,
+        1,
+        "batch",
+        telemetry_to_check=TelemetryField.PANDAS_FETCH_BATCHES,
+    )
+
+    finish(conn_cnx, table)
```

Comment:
I really like parametrized tests as it can tell us easily which one is failing when there are 2 tests that are super similar. Let me show you how it'd work in this case.
```suggestion
@pytest.mark.skipif(
    not installed_pandas or no_arrow_iterator_ext,
    reason="arrow_iterator extension is not built, or pandas is missing.",
)
@pytest.mark.parametrize("fetch_method,expected_telemetry_type",
    (
        ("one", TelemetryField.PANDAS_FETCH_ALL),
        ("batch", TelemetryField.PANDAS_FETCH_BATCHES),
    )
)
def test_pandas_telemetry(conn_cnx, capture_sf_telemetry, fetch_method, expected_telemetry_type):
    cases = ["NULL", 0.11, -0.11, "NULL", 1.27, -1.28, "NULL"]
    table = "test_pandas_telemetry"
    column = "(a number(5,2))"
    values = (
        "(" + "),(".join([f"{i}, {c}" for i, c in enumerate(cases)]) + ")"
    )
    init(conn_cnx, table, column, values)
    sql_text = f"select a from {table} order by s"

    validate_pandas_with_telemetry(
        conn_cnx,
        capture_sf_telemetry,
        sql_text,
        cases,
        1,
        fetch_method,
        telemetry_to_check=expected_telemetry_type,
    )

    finish(conn_cnx, table)
```

Reply:
> I really like parametrized tests as it can tell us easily which one is failing when there are 2 tests that are super similar. Let me show you how it'd work in this case.

It is nice but with higher cost, e.g. call init and finish twice.

Reply:
What's more important when running tests?

I'd say it's not the 5s that we save, but the error report being easier for the developers.

On the flip side, if I need to run the tests locally I can't run one without the other one.